questionnaire/questionnaire/questionnaire.js

```diff
@@ -190,6 +190,40 @@ function createQuestionnaire({
         return resolvedVars;
     }
 
+    function getDependentQuestionIds(questionId, sortingInstructions) {
+        const questionOrdering = sortingInstructions[questionId];
+        return questionOrdering || [];
+    }
+
+    function dependentsInIncorrectOrder(values, valueIndex, sortingInstructions) {
+        if (valueIndex === 0) {
+            return false;
+        }
+        const currentQuestionId = values[valueIndex].id;
+        const previousQuestionId = values[valueIndex - 1].id;
+        const dependentIds = getDependentQuestionIds(currentQuestionId, sortingInstructions);
```

Comment:
Performance Improvement: If dependentsInIncorrectOrder is a bottleneck due to a large number of dependentIds and previousQuestionIds, consider converting dependentIds to a Set for faster lookup.
const dependentIds = new Set(getDependentQuestionIds(currentQuestionId, sortingInstructions));
    return dependentIds.has(previousQuestionId);

Reply:
I don't think set conversion is necessary here as we only ever have a few dependentIds in the list (the most is for the physical injury question, which only has 33)

Reply:
agreed, just putting it out as an example and somethign to think about in the future